bn256.go

```diff
@@ -391,6 +395,10 @@ func (e *GT) Marshal() []byte {
 	// Each value is a 256-bit number.
 	const numBytes = 256 / 8
 
+	if e.p == nil {
+		e.p = &gfP12{}
+	}
+
```

Comment:
The identity value in GT is not 0, but 1.
```go
if e.p == nil {
    e.p = &gfP12{}
    G.p.SetOne()
}
```

Reply:
Yes absolutely. Wasn't sure what to do as you had not set the `e.p` to the identity in `G2`, so I followed the same approach. Makes more sense to marshal the identity of the group indeed. I saw that you added an instruction to set the identity in G2 and fixed the changes I made to the marshal functions for G1 and GT. Thanks!